### PR TITLE
Deprecate older ROCm releases - ROCm-5.1.0 till ROCm-5.2.3

### PR DESCRIPTION
**Pull request by srekolam**

### CODE REVIEW

File: var/spack/repos/builtin/packages/atmi/package.py [hunk 1]
```diff
@@ -21,15 +21,18 @@ class Atmi(CMakePackage):
...
+        version("5.4.0", sha256="b5cce10d7099fecbb40a0d9c2f29a7675315471fe145212b375e37e4c8ba5618")
+        version("5.3.3", sha256="cc1144e4939cea2944f6c72a21406b9dc5b56d933696494074c280df7469834a")
+        version("5.3.0", sha256="dffc0eb0bc1617843e7f728dbd6c8b12326c5c8baa34369aa267aab40f5deb6a")
+        version("5.2.3", sha256="5f66c59e668cf968e86b556a0a52ee0202d1b370d8406e291a874cbfd200ee17")
+        version("5.2.1", sha256="6b33445aa67444c038cd756f855a58a72dd35db57e7b63da37fe78a8585b982b")
+        version("5.2.0", sha256="33e77905a607734157d46c736c924c7c50b6b13f2b2ddbf711cb08e37f2efa4f")
+        version("5.1.3", sha256="a43448d77705b2b07e1758ffe8035aa6ba146abc2167984e8cb0f1615797b341")
+        version("5.1.0", sha256="6a758f5a8332e6774cd8e14a4e5ce05e43b1e05298d817b4068c35fa1793d333")
+
     version(
```
Comment (haampie) [inline var/spack/repos/builtin/packages/atmi/package.py:37]: Can you delete the ones that were already deprecated before the 0.21 release?

Comment (srekolam) [reply var/spack/repos/builtin/packages/atmi/package.py:37]: I am fine with either way, just keep the older SHAs if someone wants to build those releases for some reason.

**Comment by haampie** [comment]
> Cannot select a single "version" for package "rocprofiler-dev"

in CI. @srekolam maybe too much is deprecated?

**Comment by bernhardkaindl** [comment]
@srekolam

Deprecating seemingly hundreds of versions in 51 packages in a single PR is likely far too much!

@srekolam I recommend you to start very small with a very fresh PR, only very old versions and only few packages at once and only those that are really really really problematic.

**Comment by srekolam** [comment]
> @srekolam
> 
> Deprecating seemingly hundreds of versions in 51 packages in a single PR is likely far too much!
> 
> @srekolam I recommend you to start very small with a very fresh PR, only very old versions and only few packages at once and only those that are really really really problematic.

I have modified the rocm recipes and deprecated the releases from 5.1.0 to 5.2.3 . This allowed the gitlab-ci to pass now. Can you let me know if this can be merged.

**Comment by srekolam** [comment]
@tldahlgren can you please review this PR and let me know if there are any comments.

**Comment by srekolam** [comment]
@spackbot re-run pipeline

**Review by haampie — APPROVED** [review]
Thank you

**Pull request merged**